src/cartesi-machine.lua

```diff
@@ -2074,6 +2103,36 @@ if periodic_hashes_start ~= 0 then
 else
     next_hash_mcycle = periodic_hashes_period
 end
+
+-- proxy functions to snapshot/commit/rollback
+local forked_snapshot = false
+local function do_snapshot()
```

Comment:
I would simply call these snapshot/commit/rollback and pass the machine to them.
I think we are starting to use way too many global variables (e.g. cycle) that are modified inside functions. Can we try to minimize this?

Reply:
I was passing before but the `machine` word is already in use and was giving me Lua warnings, so I decided to remove, also because it will depend on the global `forked_snapshot` anyway.

Reply:
Reworked this, passing `machine` now, and I also removed `cycles` variable in favor of `machine:read_mcycle()`.